### PR TITLE
v4.0.0

### PR DESCRIPTION
**Pull request by zbrotherton**
# Pull Request Details

## Which does this affect?

ArmSubsystem - Full chain arm functionality, including feedforward and tuned PID

## Testing Done

WIP

## Checklist

- [ ] Full system testing has been done on the robot to confirm functionality/hotfix
- [x] Pull request is coming from develop or a hotfix branch

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/ArmSubsystem.java [hunk 1]
```diff
@@ -222,15 +251,26 @@ public double getChooChooAngle() {
   }
 
   public void updateReferences(double bJoystickValue, double eJoystickValue) {
-    setBaseReference(MathUtil.clamp(basePIDController.getGoal().position+Units.degreesToRadians(bJoystickValue),ArmSubsystem.kMinBaseAngle,ArmSubsystem.kMaxBaseAngle));
-    setElbowReference(MathUtil.clamp(elbowPIDController.getGoal().position+Units.degreesToRadians(eJoystickValue),ArmSubsystem.kMinElbowAngle,ArmSubsystem.kMaxElbowAngle));
+    basePIDController.setGoal(MathUtil.clamp(basePIDController.getGoal().position+Units.degreesToRadians(bJoystickValue),ArmSubsystem.kMinBaseAngle,ArmSubsystem.kMaxBaseAngle));
+    elbowPIDController.setGoal(MathUtil.clamp(elbowPIDController.getGoal().position+Units.degreesToRadians(eJoystickValue),ArmSubsystem.kMinElbowAngle,ArmSubsystem.kMaxElbowAngle));
```
Comment (azvanderpas) [inline src/main/java/frc/robot/subsystems/ArmSubsystem.java:255]: Why was this changed?

**Review by azvanderpas — APPROVED** [review]

File: src/main/java/frc/robot/subsystems/ArmSubsystem.java (hunk 1 above)
Comment (caseyjbrotherton) [reply src/main/java/frc/robot/subsystems/ArmSubsystem.java:255]: Is this for the profiled pid controller?

Just looks like it was missed before

File: src/main/java/frc/robot/subsystems/ArmSubsystem.java [hunk 2]
```diff
@@ -77,9 +78,18 @@ public double getY(){
   private double kBaseI = 0.35;
   private double kBaseD = 0;
   //Elbow arm PID values
-  private double kElbowP = 0.95;
-  private double kElbowI = 0.15;
-  private double kElbowD = 0.05;
+  private double kElbowUpP = 5;
```
Comment (caseyjbrotherton) [inline src/main/java/frc/robot/subsystems/ArmSubsystem.java:81]: Feel like this was dropped to 4.5 at the last driver practice.  Is this the current P

**Review by caseyjbrotherton — APPROVED** [review]

File: src/main/java/frc/robot/subsystems/ArmSubsystem.java (hunk 1 above)
Comment (zbrotherton) [reply src/main/java/frc/robot/subsystems/ArmSubsystem.java:255]: This prevented the arm from swapping between up/down PID when the default command started

File: src/main/java/frc/robot/subsystems/ArmSubsystem.java (hunk 2 above)
Comment (zbrotherton) [reply src/main/java/frc/robot/subsystems/ArmSubsystem.java:81]: I believe so, yes
